### ISSUE TITLE
Linux tray submenu click causes crash

### ISSUE THREAD

**Issue by jakedouglas** [nwjs/nw.js#6496]
NWJS Version : 0.28.3, 0.29.0-beta1
Operating System : Ubuntu 17.10 and other Linux

NWJS seems to crash when clicking a menu item in a tray submenu whose index is greater than or equal to the number of items in the primary menu. This only appears to happen under Linux platforms.

The following example can be used to reproduce. Note that you'll need to specify an icon file or the tray won't appear.

* Clicking anything in the primary tray menu is fine.
* Clicking `D`, `E`, `F`, and `G` under the `More` submenu are fine.
* Clicking `H` or `I` will cause NWJS to crash.

```html
<!DOCTYPE html>
<html>
  <head>
    <title>Hi</title>
  </head>
  <body style="width: 100%; height: 100%;">
    <script>
      var tray = new nw.Tray({ title: 'Tray', icon: 'path/to/icon.png' });

      var menu = new nw.Menu();
      menu.append(new nw.MenuItem({ type: 'normal', label: 'A' }));
      menu.append(new nw.MenuItem({ type: 'normal', label: 'B' }));
      menu.append(new nw.MenuItem({ type: 'normal', label: 'C' }));

      var subMenu = new nw.Menu();
      subMenu.append(new nw.MenuItem({ type: 'normal', label: 'D' }));
      subMenu.append(new nw.MenuItem({ type: 'normal', label: 'E' }));
      subMenu.append(new nw.MenuItem({ type: 'normal', label: 'F' }));
      subMenu.append(new nw.MenuItem({ type: 'normal', label: 'G' }));
      subMenu.append(new nw.MenuItem({ type: 'normal', label: 'H' }));
      subMenu.append(new nw.MenuItem({ type: 'normal', label: 'I' }));

      menu.append(new nw.MenuItem({ type: 'normal', label: 'More', submenu: subMenu }))

      tray.menu = menu;
    </script>
  </body>
</html>
```

This crashes with a message such as this:

```
[22402:22402:0306/132040.617684:FATAL:simple_menu_model.cc(438)] Check failed: static_cast<size_t>(index) < items_.size() (4 vs. 4)
#0 0x7f01b18991cc <unknown>

Aborted (core dumped)
```

I've included what seems to be the most relevant top of a stack trace from minidump using v0.28.3:

```
Thread 0 (crashed)
 0  libc-2.26.so + 0x370bb
    rax = 0x0000000000000000   rdx = 0x0000000000000000
    rcx = 0x00007f01ad9880bb   rbx = 0x00007fffde1ac7b0
    rsi = 0x00007fffde1ac4f0   rdi = 0x0000000000000002
    rbp = 0x00007fffde1ac740   rsp = 0x00007fffde1ac4f0
     r8 = 0x0000000000000000    r9 = 0x00007fffde1ac4f0
    r10 = 0x0000000000000008   r11 = 0x0000000000000246
    r12 = 0x00007fffde1acc08   r13 = 0x00007fffde1acbf8
    r14 = 0x00007fffde1acc00   r15 = 0x00007fffde1ac7b0
    rip = 0x00007f01ad9880bb
    Found by: given as instruction pointer in context
 1  libnw.so!~LogMessage [logging.cc : 844 + 0x5]
    rbp = 0x00007fffde1acbe0   rsp = 0x00007fffde1ac750
    rip = 0x00007f01b18b0e81
    Found by: previous frame's frame pointer
 2  libnw.so!ValidateItemIndex [simple_menu_model.cc : 438 + 0x28]
    rbp = 0x00007fffde1acd40   rsp = 0x00007fffde1acbf0
    rip = 0x00007f01b197be33
    Found by: previous frame's frame pointer
 3  libnw.so!GetCommandIdAt [simple_menu_model.cc : 302 + 0x5]
    rbp = 0x00007fffde1acd60   rsp = 0x00007fffde1acd50
    rip = 0x00007f01b197bfbe
    Found by: previous frame's frame pointer
 4  libnw.so!ui::SimpleMenuModel::IsEnabledAt(int) const + 0x16
    rbp = 0x00007fffde1acd90   rsp = 0x00007fffde1acd70
    rip = 0x00007f01b197c3a6
    Found by: previous frame's frame pointer
 5  libnw.so!libgtkui::AppIndicatorIconMenu::OnMenuItemActivated(_GtkWidget*) + 0xa4
    rbp = 0x00007fffde1acdc0   rsp = 0x00007fffde1acda0
    rip = 0x00007f01b3b4fc04
    Found by: previous frame's frame pointer
```

**Comment by Christywl** [comment]
I can reproduce this issue on Ubuntu 16.04 with nwjs-sdk-v0.28.3.

**Comment by rogerwang** [comment]
Will see this after 0.29.0 release, which is scheduled soon.

**Comment by jakedouglas** [comment]
Thanks. Let me know if I can do anything to help.

**Comment by rogerwang** [comment]
This is fixed in git and will be available in the next [nightly build](http://dl.nwjs.io/live-build/).

**Comment by jakedouglas** [comment]
Do you mind pointing me at the fix so I can understand? Also any idea when to expect this in a release?

Thank you!

**Comment by rogerwang** [comment]
The change is in this line: https://github.com/nwjs/chromium.src/blob/nw29/chrome/browser/ui/libgtkui/app_indicator_icon_menu.cc#L119

It will be released with 0.29.1 soon.
